mat-process-api.Tests/V1/Controllers/ProcessDataImagesControllerTests.cs

```diff
@@ -175,5 +182,329 @@ public void given_an_exception_thrown_check_that_the_controller_throws_correct_e
+        {
+            //arrange
+            GetProcessImageRequest request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockUsecase.Verify(u => u.ExecuteGet(It.IsAny<GetProcessImageRequest>()), Times.Once);
+        }
+
+        [Test]
+        public void given_a_valid_request_when_GetProcessImage_controller_method_is_called_then_it_calls_usecase_with_correct_data()
+        {
+            //arrange
+            GetProcessImageRequest request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockUsecase.Verify(u => u.ExecuteGet(It.Is<GetProcessImageRequest>(obj =>
+                obj.processRef == request.processRef &&
+                obj.imageId == request.imageId
+                )), Times.Once);
+        }
+
+        [Test]
+        public void given_a_request_when_GetProcessImage_controller_method_is_called_then_it_calls_the_validator_with_that_request_object()
+        {
+            //arrange
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+           
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //setup validator to return a no error validation result
+
+            //act
+            _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+
+            //assert
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.fileExtension == request.fileExtension)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.imageId == request.imageId)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.processRef == request.processRef)), Times.Once);
+            _mockGetValidator.Verify(v => v.Validate(It.Is<GetProcessImageRequest>(obj => obj.processType == request.processType)), Times.Once);
+        }
+
+        [Test]
+        public void when_GetProcessImage_controller_method_is_called_then_it_calls_the_logger()
+        {
+            //arrange
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
```

Comment:
'<1>'

Like before, the contents of the request object don't matter. 
It's a test about calling the validator unconditionally at least once. (like when any request comes in - even if it's empty)

> This is a nitpicky thing, but it helps a ton, when reading tests. I've seen there were a lot of changes from:
> `var request = new GetProcessImageRequest();`, to:
> `var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();`
> 
> The reason why this line: `var request = new GetProcessImageRequest();` was left as is rather than changed to be generated from helper was that it helps to signify that this test does not depend on what the request object is, and that the only reason the request object is there is to keep the compiler happy so that the test could be run.
> 
> Now that it's changed the helper, it's no longer clear if the contents of the request object are of any significance to the test. Also the line of code is much longer now with a lot more happening, which makes it visually heavier to read in the future.